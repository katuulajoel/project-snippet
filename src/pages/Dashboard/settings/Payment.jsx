import React from "react";
import { useSelector } from "react-redux";
import { ENDPOINT_PAYONEER_SIGNUP } from "../../../utils/api";
// import { STATUS_INITIATED } from "../../../configs/constants/ActionTypes";
import styled from "styled-components";
// export const STATUS_APPROVED = "approved";
// export const STATUS_DECLINED = "declined";
export const STATUS_INITIATED = "initiated";
// export const STATUS_PENDING = "pending";

const Payment = () => {
  const { user } = useSelector(({ Auth }) => Auth);

  return (
    <div>
      <PayoneerCard>
        <div>
          <h4>Payoneer</h4>
          <p>
            To receive payments for your work on Tunga, you need to setup
            Payoneer. You can do so using the button below.
          </p>
          <h5>Status: {user.payoneer_status || ""}</h5>

          <a
            href={`${ENDPOINT_PAYONEER_SIGNUP}?next_url=${encodeURIComponent(
              user.payoneer_signup_url
            )}&error_url=${encodeURIComponent(
              `${window.location.origin}/settings/payment`
            )}`}
            className="btn btn-primary"
            title="Setup Payoneer"
          >
            {user && user.payoneer_status === STATUS_INITIATED
              ? "Having trouble? Re-setup Payoneer account"
              : "Setup Payoneer"}
          </a>
        </div>
      </PayoneerCard>
    </div>
  );
};

const PayoneerCard = styled.div`
  h4 {
    font-weight: 600;
    font-size: 18px;
    line-height: 22px;
    /* identical to box height */

    color: #151a30;
  }
  a {
    box-shadow: none !important;
  }
  p {
    font-weight: 300;
    font-size: 16px;
    line-height: 150%;
    /* or 24px */

    color: #3e4857;
  }
`;

export default Payment;
