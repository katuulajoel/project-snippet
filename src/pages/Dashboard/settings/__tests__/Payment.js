import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import Payment from "../Payment";

describe("Auth layout test", () => {
  it("Snapshot test for Login component", () => {
    const tree = renderer
      .create(
        <Router>
          <Payment />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
