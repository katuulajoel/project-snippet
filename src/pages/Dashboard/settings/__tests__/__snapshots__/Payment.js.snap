// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth layout test Privacy component snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="sc-bdnxRM bzbqFw"
    >
      <div>
        <h4>
          Payoneer
        </h4>
        <p>
          To receive payments for your work on Tunga, you need to setup Payoneer. You can do so using the button below.
        </p>
        <h5>
          Status: 
        </h5>
        <a
          class="btn btn-primary"
          href="undefinedapi/payoneer/?next_url=undefined&error_url=http%3A%2F%2Flocalhost%2Fsettings%2Fpayment"
          title="Setup Payoneer"
        >
          Setup Payoneer
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
